libparsec/crates/platform_device_loader/src/lib.rs

```diff
@@ -198,19 +196,40 @@ impl From<SaveDeviceError> for ChangeAuthentificationError {
 }
 
 /// Note `config_dir` is only used as discriminant for the testbed here
-pub async fn change_authentication(
+pub async fn update_device_change_authentication(
     #[cfg_attr(not(feature = "test-with-testbed"), allow(unused_variables))] config_dir: &Path,
     current_access: &DeviceAccessStrategy,
     new_access: &DeviceAccessStrategy,
-) -> Result<AvailableDevice, ChangeAuthentificationError> {
+) -> Result<AvailableDevice, UpdateDeviceError> {
+    #[cfg(feature = "test-with-testbed")]
+    if let Some(result) = testbed::maybe_update_device(config_dir, current_access, new_access, None)
+    {
+        return result.map(|(available_device, _)| available_device);
+    }
+
+    platform::update_device(current_access, new_access, None)
+        .await
+        .map(|(available_device, _)| available_device)
+}
+
+/// Note `config_dir` is only used as discriminant for the testbed here
+///
+/// Returns the old server address
```

Comment:
Not sure if this is relevant but there seem to be an asymmetry between
- `update_device_change_authentication` returning the updated device, and
- `update_device_overwrite_server_address` returning the old server address

Why not returning the device in both cases? or the changed element in both cases?
Also, why "overwrite" and not simply "change"?

There is beauty in symmetry ^^

Reply:
https://github.com/Scille/parsec-cloud/pull/10017#discussion_r2018511704

Reply:
Returning `AvailableDevice` is useful to display the new device (e.g. showing its short ID in the CLI, displaying the organization & human handle when importing a recovery device)

However in case of `update_device_overwrite_server_address` we don't care about this (since those information don't change at all), but want something totally unrelated (i.e. the old server url)